tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,845 @@
+	labelKey = "scenario"
+
+	getApp := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: application(id: %s,) {
+						%s
+					}
+				}`,
+			firstApp.ID, tc.gqlFieldsProvider.ForApplication()))
+
+	app := graphql.Application{}
+	// WHEN
+	err = tc.RunQuery(ctx, getApp, app)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, app)
+	assert.Contains(t, app.Labels, labelKey)
+	assert.Equal(t, "default", app.Labels[labelKey])
+}
+
+func TestDeleteLabelDefinition(t *testing.T) {
+	//Create application
+	// GIVEN
+	ctx := context.Background()
+
+	in := generateSampleApplicationInput("first")
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+
+	createReq := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+  				result: createApplication(in: %s) {
+    					%s
+					}
+				}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	firstApp := graphql.Application{}
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &firstApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Create LabelDefinition
+	// GIVEN
+	labelKey := "foo"
+	jsonSchema := `{
+	  "$id": "https://foo.com/bar.schema.json",
+	  "title": "foobarbaz",
+	  "type": "object",
+	  "properties": {
+		"foo": {
+		  "type": "string",
+		  "description": "foo"
+		},
+	  },
+      "required": ["foo"]
+}`
+
+	schema := interface{}(&jsonSchema)
+	labelDefinitionInput := graphql.LabelDefinitionInput{
+		Key:    labelKey,
+		Schema: &schema,
+	}
+
+	ldInputGql, err := tc.graphqlizer.LabelDefinitionInputToGQL(labelDefinitionInput)
+	require.NoError(t, err)
+
+	createLabelDefinitionRequest := gcli.NewRequest(
+		//    createLabelDefinition(in: LabelDefinitionInput!): LabelDefinition!
+		fmt.Sprintf(`mutation {
+				result: createLabelDefinition(in: %s) {
+						%s
+					}
+				}`,
+			ldInputGql, tc.gqlFieldsProvider.ForLabelDefinition()))
+	ld := graphql.LabelDefinition{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, createLabelDefinitionRequest, ld)
+
+	// THEN
+	require.NoError(t, err)
+	assert.Equal(t, labelKey, ld.Key)
+
+	//Set label on application
+	// GIVEN
+	validLabelValue := "test"
+
+	setLabelRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: setApplicationLabel(applicationID: %s, key: %s, value: %v) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKey, validLabelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+	label := graphql.Label{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, setLabelRequest, &label)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, label.Key)
+	require.NotEmpty(t, label.Value)
+
+	// Try to delete labeldefinition when it's used by some labels
+	// GIVEN
+	queryAppReq := gcli.NewRequest(
+		fmt.Sprintf(`query {
+			result: deleteLabelDefinition(key: "%s") {
+					%s
+				}
+			}`, labelKey, tc.gqlFieldsProvider.ForLabelDefinition()))
+
+	// WHEN
+	err = tc.RunQuery(context.Background(), queryAppReq, nil)
+
+	//THEN
+	require.Error(t, err)
```

Comment:
I think we should check what's the error

Reply:
added missing error checks